Include/internal/pycore_dict_state.h

```diff
@@ -8,33 +8,14 @@ extern "C" {
 #  error "this header requires Py_BUILD_CORE define"
 #endif
 
-
-#ifndef WITH_FREELISTS
-// without freelists
-#  define PyDict_MAXFREELIST 0
-#endif
-
-#ifndef PyDict_MAXFREELIST
-#  define PyDict_MAXFREELIST 80
-#endif
-
 #define DICT_MAX_WATCHERS 8
 
-struct _Py_dict_state {
+struct _Py_dict_interp_state {
```

Comment:
The pattern we've been following for runtime state sub-struct names is:

* `_Py_TYPE_state` for interpreter state
* `_Py_TYPE_runtime_state` for global runtime state

Let's not introduce a new naming scheme.

(If we change the freelists struct name as we have discussed there is no longer any need to avoid a name collision.)

Reply:
Done! :)

For the rest of the structs, I will rename them in a one-shot PR.
Also separating _Py_dict_freelist, we can handle it as the separated PR.